core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -530,7 +542,7 @@ private[spark] class TaskSetManager(
 
   private def maybeFinishTaskSet() {
     if (isZombie && runningTasks == 0) {
-      sched.taskSetFinished(this)
+      sched.taskSetFinished(this, tasksSuccessful == numTasks)
```

Comment:
but you said only active task set manager can clean up the `stageIdToFinishedPartitions`? This branch is under `if (isZombie)`